addon/package.json

```diff
@@ -9,11 +9,32 @@
   "repository": "https://github.com/ember-modifier/ember-modifier",
   "license": "MIT",
   "author": "",
-  "types": "index.d.ts",
+  "exports": {
+    ".": "./dist/index.js",
+    "./*": "./dist/*",
+    "./addon-main.js": "./addon-main.js"
+  },
+  "typesVersions": {
+    "*": {
+      ".": [
+        "./dist/index.d.ts"
+      ],
+      "*": [
+        "./dist/*"
+      ],
+      "./-private/signature": [
+        "./dist/-private/signature.d.d.ts"
+      ]
+    }
+  },
```

Comment:
Overall I'm fine with landing this as is for now, but we may want to revisit. Among other things, it'd be good not to expose the `-private` bit *at all*.

Reply:
@chriskrycho the only reason for exposing `-private/signature` is used in tests to import `DefaultSignature, NamedArgs, PositionalArgs` and seem like `DefaultSignature` is not available for import from `ember-modifier`.

Do you think we can make `DefaultSignature` importable?

Reply:
We should not make it importable, and we should use *import* maps to make the others available for import instead!

Reply:
Again, though, it's fine to land this way for now; we can tweak it after landing the PR!

Reply:
@chriskrycho agree! there is always room for improvement :)